dnf/cli/cli.py

```diff
@@ -725,6 +725,10 @@ def _process_demands(self):
             for repo in repos.iter_enabled():
                 repo.load_metadata_other = True
 
+        if 'filelists' in self.base.conf.optional_metadata_types:
```

Comment:
I have a bad news. This will not work for API users. 

```
base = dnf.base.Base()

base.repos.add_new_repo("fedora", base.conf, metalink="https://mirrors.fedoraproject.org/metalink?repo=fedora-38&arch=x86_64")

base.fill_sack(load_system_repo=False)
```
I am looking for a new location.

Reply:
I have two suggestion. I believe that the best option to synchronize it would be in `Repo` Class in `def load(self):` as a first item to perform. Repo class has an access to RepoConfig and it should have an access to MainConfig (`_config` or `_mainConfigRefHolder` attributes). 

The second place would be in `Base._add_repo_to_sack(self, repo):` as a first item to perform. This place will not help projects that used low level API (`Repo Class def load()`).

Reply:
I've chosen the `Base._add_repo_to_sack` place as it seems to me we should allow the API users calling the methods at the `Repo` level manually to setup/override the filelists load request during the runtime.